### PR TITLE
Move CI documentation to inside Breeze docs

### PR DESCRIPTION
**Pull request by potiuk**
This PR moves the documentation of CI of ours to inside Breeze doc folder and splits the documentation in separate docs / chapters following similar changes done for Breeze docs #36936 and the contributing docs #36969.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

**Review by dirrao — APPROVED** [review]

File: dev/breeze/doc/ci/01_ci_environment.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+- [CI Environment](#ci-environment)
+  - [GitHub Actions runs](#github-actions-runs)
+  - [Container Registry used as cache](#container-registry-used-as-cache)
+  - [Authentication in GitHub Registry](#authentication-in-github-registry)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# CI Environment
+
+Continuous Integration is important component of making Apache Airflow
```
Comment (aritra24) [inline dev/breeze/doc/ci/01_ci_environment.md]: Not: is an* important component

File: dev/breeze/doc/ci/01_ci_environment.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+robust and stable. We are running a lot of tests for every pull request,
```
Comment (aritra24) [inline dev/breeze/doc/ci/01_ci_environment.md]: We run* a lot of tests

File: dev/breeze/doc/ci/01_ci_environment.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+for main and v2-\*-test branches and regularly as scheduled jobs.
+
+Our execution environment for CI is [GitHub
+Actions](https://github.com/features/actions). GitHub Actions (GA) are
+very well integrated with GitHub code and Workflow and it has evolved
+fast in 2019/202 to become a fully-fledged CI environment, easy to use
```
Comment (aritra24) [inline dev/breeze/doc/ci/01_ci_environment.md]: 2019/2020* to become

File: dev/breeze/doc/ci/01_ci_environment.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+## Container Registry used as cache
+
+We are using GitHub Container Registry to store the results of the
+`Build Images` workflow which is used in the `Tests` workflow.
+
+Currently in main version of Airflow we run tests in all versions of
+Python supported, which means that we have to build multiple images (one
+CI and one PROD for each Python version). Yet we run many jobs (\>15) -
+for each of the CI images. That is a lot of time to just build the
+environment to run. Therefore we are utilising `pull_request_target`
```
Comment (aritra24) [inline dev/breeze/doc/ci/01_ci_environment.md]: utilising the* `pull_request_target`

File: dev/breeze/doc/ci/01_ci_environment.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+feature of GitHub Actions.
+
+This feature allows to run a separate, independent workflow, when the
```
Comment (aritra24) [inline dev/breeze/doc/ci/01_ci_environment.md]: allows us* to run

**Review by aritra24 — COMMENTED** [review]
A few nits on the first walk through, overall lgtm

File: dev/breeze/doc/ci/01_ci_environment.md (hunk 3 above)
Comment (potiuk) [reply dev/breeze/doc/ci/01_ci_environment.md]: removed it altogether and updated. It's prehistory, not relevant any more.

**Review by aritra24 — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**